libs/core/src/lib/message-strip/message-strip.component.ts

```diff
@@ -0,0 +1,121 @@
+import { ChangeDetectionStrategy, Component, ElementRef, Input, OnChanges, OnInit, ViewEncapsulation, EventEmitter, Output } from '@angular/core';
+import { applyCssClass, CssClassBuilder } from '../utils/public_api';
+
+let messageStripUniqueId: number = 0;
+
+/**
+ * The component that represents a message-strip. It can only be used inline.
+ */
+@Component({
+    selector: 'fd-message-strip',
+    templateUrl: './message-strip.component.html',
+    styleUrls: ['./message-strip.component.scss'],
+    host: {
+        '[attr.aria-labelledby]': 'ariaLabelledBy',
+        '[attr.aria-label]': 'ariaLabel',
+        '[style.width]': 'width',
+        '[style.min-width]': 'minWidth',
+        'role': 'alert',
+        '[attr.id]': 'id',
+    },
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MessageStripComponent implements OnInit, OnChanges, CssClassBuilder {
+
+    /** user's custom classes */
+    @Input()
+    class: string = '';
+
+    /** Whether the message strip is dismissible. */
+    @Input()
+    dismissible: boolean = true;
+
+    /** The default message strip does not have an icon. 
+     * The other types (warning, success, information and error) have icons by default. 
+     * To remove the icon set the property to true. 
+     */
+    @Input()
+    noIcon: boolean = false;
+
+    /** The type of the message strip. 
+     * Can be one of *warning*, *success*, *information*, *error* or null. 
+     */
+    @Input()
+    type: string;
+
+    /** Id for the message-strip component. If omitted, a unique one is generated. */
+    @Input()
+    id: string = 'fd-message-strip-' + messageStripUniqueId++;
+
+    /** Id of the element that labels the message-strip. */
+    @Input()
+    ariaLabelledBy: string = null;
+
+    /** Aria label for the message-strip component element. */
+    @Input()
+    ariaLabel: string = null;
+
+    /** Aria label for the dismiss button. */
+    @Input()
+    dismissLabel: string = 'Dismiss';
+
+    /** Width of the message-strip. */
+    @Input()
+    width: string;
+
+    /** Minimum width of the message-strip. */
+    @Input()
+    minWidth: string;
+
+    /** Alternative way of passing in a message to the message-strip. */
+    @Input()
+    message: string;
+
+    /** Event fired when the message-strip is dismissed. */
+    @Output()
+    onDismiss: EventEmitter<undefined> = new EventEmitter<undefined>();
```

Comment:
Its barely no difference, but I think we should follow current convention and go with:
```
    onDismiss: EventEmitter<void> = new EventEmitter<void>();
```

Reply:
Fixed